core/unit_test/TestPolicyConstruction.hpp

```diff
@@ -562,6 +712,164 @@ class TestTeamPolicyConstruction {
     }
   }
 
+  void test_compile_time_deduction_guides() {
+    Kokkos::DefaultExecutionSpace des{};
+    ImplicitlyConvertibleToDefaultExecutionSpace notEs{};
+    SomeExecutionSpace ses{};
+    ExecutionSpace es{};
+
+    using PolicyExecSpace = std::conditional_t<
+        std::is_same_v<ExecutionSpace, Kokkos::DefaultExecutionSpace>,
+        Kokkos::TeamPolicy<>, Kokkos::TeamPolicy<ExecutionSpace>>;
+
+    int i{};
```

Comment:
Setting the team size to 0 causes some of the test failures.

Reply:
Changed the team size to 1.